flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -432,6 +434,20 @@ private StreamConfig createJobVertex(
 					chainedNames.get(streamNodeId),
 					jobVertexId,
 					operatorIDPairs);
+			if (streamNode.getOperatorFactory() instanceof CoordinatedOperatorFactory) {
+				OperatorCoordinator.Provider coordinatorProvider =
+						((CoordinatedOperatorFactory<?>) streamNode
+								.getOperatorFactory())
+								.getCoordinatorProvider(
+										chainedNames.get(streamNodeId),
+										new OperatorID(hashes.get(streamNodeId)));
+				try {
+					jobVertex.addOperatorCoordinator(new SerializedValue<>(coordinatorProvider));
+				} catch (IOException e) {
+					throw new FlinkRuntimeException(String.format(
+							"Coordinator Provider for node %s is not serializable.", chainedNames.get(streamNodeId)));
+				}
+			}
```

Comment:
This will only add the coordinator of the head of the operator chain into the job vertex, coordinators of other operators will be lost.

Reply:
I think @TsReaper's comment here is correct.

The proposed solution is a bit hacky, but a clean solution would need some major prerequisite cleanup of the whole StreamGraph, and StreamingJobGraphGenerator, which would be a separate bigger effort.